src/main/java/cloud/eppo/ConfigurationRequestor.java

```diff
@@ -54,23 +49,32 @@ public CompletableFuture<Void> setInitialConfiguration(
       throw new IllegalStateException("Configuration future has already been set");
     }
     this.configurationFuture =
-        configurationFuture.thenAccept(
-            (config) -> {
-              synchronized (configurationStore) {
-                if (config == null) {
-                  log.debug("Initial configuration future returned null");
-                } else if (remoteFetchFuture != null
-                    && remoteFetchFuture.isDone()
-                    && !remoteFetchFuture.isCompletedExceptionally()) {
-                  // Don't clobber a successful fetch.
-                  log.debug("Fetch successfully beat the initial config; not clobbering");
-                } else {
-                  log.debug("saving initial configuration");
-                  configurationStore.saveConfiguration(config);
-                  initialConfigSet = true;
-                }
-              }
-            });
+        configurationFuture
+            .thenAccept(
+                (config) -> {
+                  synchronized (configurationStore) {
+                    if (config == null || config.isEmpty()) {
+                      log.debug("Initial configuration future returned empty/null");
+                    } else if (remoteFetchFuture != null
+                        && remoteFetchFuture.isDone()
+                        && !remoteFetchFuture.isCompletedExceptionally()) {
+                      // Don't clobber a successful fetch.
+                      log.debug("Fetch successfully beat the initial config; not clobbering");
```

Comment:
I'm not sure this is the best error message. Perhaps something like; "Fetch has completed; ignoring loaded initial configuration"